src/background.ts

```diff
@@ -138,6 +138,28 @@ protocol.registerSchemesAsPrivileged([
 
 const firstUrl = process.env.VITE_DEV_SERVER_URL ?? "app://./index.html";
 
+const tempProjectPath = path.join(fixedUserDataDir, "project.tmp.json");
+
+const setTempProject = (tempProject: ArrayBuffer) => {
+  try {
+    fs.writeFileSync(tempProjectPath, new DataView(tempProject));
```

Comment:
fs.promisesでも良さそう。
あと、PCクラッシュの復元の場合、書き込み中にPCが死ぬと復元できなくなるので、

1. project.tmp.jsonを別ファイルにコピー（以下仮オートセーブ）
2. project.tmp.jsonに書き込み
（3. 仮オートセーブを削除）

読み込み側：
1. project.tmp.jsonを読み込み
2. 失敗したら仮オートセーブを読み込み

ってやると良さそうな気がします。

Reply:
あるいは、確実に`project.tmp.json`に正しいプロジェクトファイルが保存されている状態にしてもいいかもです！
違うファイルに書き出したrenameすればアトミックにファイル更新できるはず。雑なコードですが例えばこんな感じで！

```suggestion
    const tempTempPath = tempProjectPath + ".tmp" // ちなみに同じドライブに書き込まれてることが大事
    fs.writeFileSync(tempTempPath, new DataView(tempProject));
    fs.renameSync(tempTempPath, tempProjectPath)
```